kmip/services/server/server.py

```diff
@@ -287,17 +287,26 @@ def interrupt_handler(trigger, frame):
         for cipher in auth_suite_ciphers:
             self._logger.debug(cipher)
 
-        self._socket = ssl.wrap_socket(
+        cafile = self.config.settings.get('ca_path')
+        context = ssl.SSLContext(self.auth_suite.protocol)
+        context.verify_mode = ssl.CERT_REQUIRED
+        if self.auth_suite.ciphers:
+            context.set_ciphers(self.auth_suite.ciphers)
+        if cafile:
+            context.load_verify_locations(cafile)
+        certfile = self.config.settings.get('certificate_path')
+
+        if certfile:
+            keyfile = self.config.settings.get('key_path')
+            context.load_cert_chain(certfile, keyfile=keyfile)
+        else:
+            raise ValueError("certfile must be specified for server-side operations")
```

Comment:
What we don't have here is the equivalent of the client-side check

```python
        if self.keyfile and not self.certfile:
            raise ValueError("certfile must be specified")
```

but it also turns out that the higher levels already check that both certificate and key are specified:

```
kmip.core.exceptions.ConfigurationError: Setting 'key_path' is missing from the configuration file.
kmip.core.exceptions.ConfigurationError: Setting 'certificate_path' is missing from the configuration file.
```

So it's probably not necessary.

Reply:
Also, it looks like this causes the test suite to fail in `TestKmipServer.test_start`:

```
ValueError: certfile must be specified for server-side operations
```

So either this code or the test setup would need adjustments.